include/libtorrent/aux_/io.hpp

```diff
@@ -50,12 +51,12 @@ namespace libtorrent { namespace aux {
 	read_impl(span<Byte>& view, type<T>)
 	{
 		T ret = 0;
-		for (std::size_t i = 0; i < sizeof(T); ++i)
+		for (int i = 0; i < int(sizeof(T)); ++i)
```

Comment:
What's the point of this change?

Reply:
I bet it's to avoid the cast at `ret |= static_cast<std::uint8_t>(view[i]);` (but I know, two new casts instead)

Reply:
Precisely.

Reply:
in my mind `sizeof()` is the operator at fault here. It should really return `std::ptrdiff_t`. I think this makes that clearer.
Btw, the second cast (on line 59) would have to be there regardless of whether `i` is signed or not. So changing the type of the counter does not introduce a net new cast, it just moves one.

It makes me wonder if it would make sense, in the future, to do something like:
```
template <typename T>
inline std::ptrdiff_t sizeof_v(sizeof(T));
```
(with some appropriate static asserts perhaps)

In addition to that, I prefer [signed loop counters](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es107-dont-use-unsigned-for-subscripts-prefer-gslindex).

Perhaps this would be better as a range-for loop though, over: `view.first(sizeof(T))`